test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -423,22 +424,131 @@ class test_ucp_tag_limits : public test_ucp_tag {
 public:
     test_ucp_tag_limits() {
         m_test_offload = get_variant_value();
-        m_env.push_back(new ucs::scoped_setenv("UCX_RC_TM_ENABLE",
-                                               ucs::to_string(m_test_offload).c_str()));
+        if (m_test_offload) {
+            m_env.push_back(new ucs::scoped_setenv("UCX_RC_TM_ENABLE", "y"));
+        }
+        m_min_rndv = 0;
     }
 
     void init() {
+        /* TODO: Currently all the tests are for intra-node communication only.
```

Comment:
WDYT, would the following way help to imitate inter-node testing?

```c
    ucp_worker_cfg_index_t ep_cfg_index                  = sender().ep()->cfg_index;
    ucs_array_elem(&worker->ep_config, ep_cfg_index).key = 0;
```

Reply:
Thanks for the hint, will try that

Reply:
No, it does not build, but if it does, it will be useless.

The problem is that code is pretty much tightly coupled in that part.
The interesting part happens in `ucp_wireup_init_lanes` function that does:
```
ucp_wireup_select_lanes
  ucp_wireup_construct_lanes
    ucp_wireup_select_set_locality_flags
      assign UCP_EP_CONFIG_KEY_FLAG_INTRA_NODE
```
After that the EP is considered as local
And soon after that, in the same function we initialize the config, that discovers all the protocols: `ucp_worker_get_ep_config`.

So if we want to mock inter-node communication, we need to change ep config between these 2 calls.
How? I see some alternatives
1. play with `ucp_wireup_check_config_intersect` which is in the middle?
2. introduce new config param, e.g. force_inter
3. use mock library/framework. e.g. cpputest
4. mock at linker level, e.g. by using weak link for `ucp_wireup_check_config_intersect` function

So far I don't see an easy way to mock inter-node communication, I've already spent few days in November trying to mock it, but no luck. For now we agreed with Mikhail that I don't spend more time on this, I just tested inter-node communication manually. But IMO this might be useful in other places too